File: src/main/java/sample/Department.java
package sample;

import org.seasar.doma.Entity;
import org.seasar.doma.Id;
import org.seasar.doma.Version;

@Entity(immutable = true)
public class Department {
  @Id private final Integer id;
  private final String name;
  @Version private final Integer version;

  public Department(Integer id, String name, Integer version) {
    this.id = id;
    this.name = name;
    this.version = version;
  }

  public Integer getId() {
    return id;
  }

  public String getName() {
    return name;
  }

  public Integer getVersion() {
    return version;
  }
}

File: src/main/kotlin/sample/Department.kt
package sample

import org.seasar.doma.Entity
import org.seasar.doma.Id
import org.seasar.doma.Metamodel
import org.seasar.doma.Version

@Entity(immutable = true, metamodel = Metamodel())
class Department(@Id val id: Int, val name: String, @Version val version: Int)